subprojects/composite-builds/src/integTest/groovy/org/gradle/integtests/composite/CompositeBuildBuildPathAssignmentIntegrationTest.groovy

```diff
@@ -182,22 +213,34 @@ class CompositeBuildBuildPathAssignmentIntegrationTest extends AbstractComposite
         fixture.progress(BuildIdentifiedProgressDetails).findAll { it.details.buildPath }*.details*.buildPath
     }
 
-    private void configureBuildConfigurationOutput(List<BuildTestFile> builds) {
+    private void configureNestedBuild(List<BuildTestFile> builds) {
         builds.each {
-            it.buildFile << """
-                    subprojects {
-                        // The Java plugin forces the configuration of the included builds
-                        apply plugin: 'java'
+            it.settingsFile << """
+                gradle.projectsEvaluated {
```

Comment:
❓ I think this won't work with configuration caching, since the `projectsEvaluated` hook doesn't fire, right? Is there a way to do this in a configuration cache compatible way?

Reply:
It seems this works as expected, possibly because we run the build only once.